User Tasks: services and clients implementation

marcoandredinis opened the pull request:
This PR adds the implementation for the User Tasks:
- services (backend+cache)
- clients (API + `tctl`)
- light validation to set up the path for later PRs

Context: https://github.com/gravitational/teleport/issues/41909

r0mant reviewed (approved):
This is just boilerplate so approving but before moving forward with this please address feedback on the proto definition PR: https://github.com/gravitational/teleport/pull/46059#pullrequestreview-2277900867.

espadolini commented on lib/services/local/user_integration_task.go:
```diff
@@ -0,0 +1,85 @@
+}
+
+func (s *UserIntegrationTasksService) UpsertUserIntegrationTask(ctx context.Context, userIntegrationTask *userintegrationtasksv1.UserIntegrationTask) (*userintegrationtasksv1.UserIntegrationTask, error) {
+	r, err := s.service.UpsertResource(ctx, userIntegrationTask)
```
Don't we usually have validation checks at the `lib/services/local` layer rather than the auth service layer?

marcoandredinis replied:
I think we use both places
I did the boilerplate code by following a resource that was validating at the auth service layer.

Let me know if you think this should be moved.
I don't think there's much of a difference given that the method ends up being called right away.

espadolini replied:
If you leave it like this you're likely not going to get validation when putting values in or pulling values out from the cache (if this ends up being cached in the future), which is atypical, at the very least - it might be correct when pulling values out, but we should probably check them before storing them locally.

marcoandredinis replied:
Moved to the backend layer at `lib/services/local/user_task.go`

marcoandredinis commented on api/types/userintegrationtasks/object.go:
```diff
@@ -0,0 +1,97 @@
+// ValidateUserIntegrationTask validates the UserIntegrationTask object without modifying it.
+func ValidateUserIntegrationTask(uit *userintegrationtasksv1.UserIntegrationTask) error {
+	switch {
+	case uit == nil:
```
Changed to use getters 👍

marcoandredinis commented:
@espadolini Can you please take another look?

marcoandredinis committed:
User Tasks: services and clients implementation

This PR adds the implementation for the User Tasks:
- services (backend+cache)
- clients (API + tctl)
- light validation to set up the path for later PRs
